pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -78,23 +94,88 @@ func NewInstallerController(
 	return c
 }
 
+func (c *InstallerController) getStaticPodState(nodeName string) (state staticPodState, deploymentID string, errors []string, err error) {
+	pod, err := c.kubeClient.CoreV1().Pods(c.targetNamespace).Get(c.staticPodName+"-"+nodeName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return staticPodStatePending, "", nil, nil
+		}
+		return staticPodStatePending, "", nil, err
+	}
+	switch pod.Status.Phase {
+	case corev1.PodRunning, corev1.PodSucceeded:
+		for _, c := range pod.Status.Conditions {
+			if c.Type == corev1.PodReady && c.Status == corev1.ConditionTrue {
+				return staticPodStateReady, pod.Labels[deploymentIDLabel], nil, nil
```

Comment:
check this label exists?

Reply:
why? This returns `""` which is correct.